# Research Question 1 (RQ1): Does the level of parental education (Medu and Fedu) significantly impact students' final grades (G3)?



### Null Hypothesis (H0): The level of parental education does not significantly impact students' final grades.

### Alternative Hypothesis (H1): The level of parental education significantly impacts students' final grades.

### Test: Multiple Linear Regression



In [11]:
import pandas as pd
import statsmodels.api as sm
from scipy import stats
from scipy.stats import chi2_contingency

In [5]:
df = pd.read_csv('student-por.csv')

df.head()

,school,sex,age,address,famsize,Pstatus,Medu,Fedu,Mjob,Fjob,...,famrel,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3
0,GP,F,18,U,GT3,A,4,4,at_home,teacher,...,4,3,4,1,1,3,4,0,11,11
1,GP,F,17,U,GT3,T,1,1,at_home,other,...,5,3,3,1,1,3,2,9,11,11
2,GP,F,15,U,LE3,T,1,1,at_home,other,...,4,3,2,2,3,3,6,12,13,12
3,GP,F,15,U,GT3,T,4,2,health,services,...,3,2,2,1,1,5,0,14,14,14
4,GP,F,16,U,GT3,T,3,3,other,other,...,4,3,2,1,2,5,0,11,13,13


In [6]:
# Select the relevant columns for the analysis
data = df[['Medu', 'Fedu', 'G3']]

# Add a constant (intercept) term for the linear regression model
data['intercept'] = 1

# Define the independent variables (parental education) and the dependent variable (final grades)
X = data[['Medu', 'Fedu', 'intercept']]
y = data['G3']

# Fit the multiple linear regression model
model = sm.OLS(y, X).fit()

# Print the summary statistics of the regression model
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                     G3   R-squared:                       0.063
Model:                            OLS   Adj. R-squared:                  0.060
Method:                 Least Squares   F-statistic:                     21.77
Date:                Wed, 18 Oct 2023   Prob (F-statistic):           7.12e-10
Time:                        12:40:35   Log-Likelihood:                -1660.3
No. Observations:                 649   AIC:                             3327.
Df Residuals:                     646   BIC:                             3340.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Medu           0.5051      0.142      3.550      0.0

/var/folders/t9/xzq_p4jn637gk3j8dlymkptc0000gn/T/ipykernel_95173/1037272777.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['intercept'] = 1


## Conclusions

The coefficient for Medu (mother's education) is statistically significant (p < 0.05), indicating that it has a significant impact on students' final grades.

The coefficient for Fedu (father's education) has a p-value of 0.053, which is very close to the 0.05 threshold. While it's not statistically significant at the 5% level, it is still worth considering. Depending on the context and your research objectives, you might choose to interpret it cautiously and investigate further.

Keep in mind that the choice of the significance level (e.g., 0.05, 0.01) can vary depending on the specific requirements of your analysis or research field. If a lower significance level is used (e.g., 0.01), Fedu may not be considered statistically significant.

In summary, the output indicates that the model suggests a statistically significant relationship between maternal education levels (Medu) and students' final grades (G3). However, the R-squared value is relatively low, suggesting that these variables explain only a small portion of the variation in final grades.

Therefore, you would reject the null hypothesis (H0) for Medu and conclude that Medu has a statistically significant impact on students' final grades.

We fail to reject the null hypothesis for Fedu, suggesting that there is not enough evidence to conclude that Fedu has a statistically significant impact on students' final grades. 



# Research Question 2 (RQ2): Is there a significant difference in final grades (G3) between students who receive additional educational support (schoolsup) and those who do not?


### Null Hypothesis (H0): There is no significant difference in final grades between students who receive additional educational support and those who do not.
### Alternative Hypothesis (H1): There is a significant difference in final grades between students who receive additional educational support and those who do not.
### Test: Independent Samples T-Test

In [10]:

# Define the two groups based on the 'schoolsup' variable
group1 = df[df['schoolsup'] == 'yes']['G3']  # Group of students who receive additional support
group2 = df[df['schoolsup'] == 'no']['G3']   # Group of students who do not receive additional support

# Perform an Independent Samples T-Test
t_statistic, p_value = stats.ttest_ind(group1, group2)

# Define the significance level (alpha)
alpha = 0.05

# Print the results and make a significance decision
print("T-Statistic:", t_statistic)
print("P-Value:", p_value)

if p_value < alpha:
    print("Result: Reject the null hypothesis (H0)")
    print("Conclusion: There is a significant difference in final grades between students who receive additional educational support and those who do not.")
else:
    print("Result: Fail to reject the null hypothesis (H0)")
    print("Conclusion: There is no significant difference in final grades between the two groups.")


T-Statistic: -1.6928205340484455
P-Value: 0.09097103846579743
Result: Fail to reject the null hypothesis (H0)
Conclusion: There is no significant difference in final grades between the two groups.


## Conclusions

The calculated t-value is approximately -1.693.
The p-value associated with the t-test is approximately 0.091.
Since the t-value is negative, it suggests that the sample mean of the group receiving additional educational support is slightly smaller than the sample mean of the group not receiving support. However, because p-value is 0.091, we fail to reject the null hypothesis. This indicates that the observed difference in means is not statistically significant, and any observed differences could have occurred by chance.






# Research Question 3 (RQ3): Is there a significant association between students' internet access at home (internet) and their desire to pursue higher education (higher)?


### Null Hypothesis (H0): There is no significant association between students' internet access at home and their desire to pursue higher education.

### Alternative Hypothesis (H1): There is a significant association between students' internet access at home and their desire to pursue higher education.

### Test: Chi-Square Test of Independence

In [12]:

# Create a contingency table by cross-tabulating 'internet' and 'higher'
contingency_table = pd.crosstab(df['internet'], df['higher'])

# Perform the chi-square test of independence
chi2, p_value, _, _ = chi2_contingency(contingency_table)

# Define the significance level (alpha)
alpha = 0.05

# Print the results and make a significance decision
print("Chi-Square Statistic:", chi2)
print("P-Value:", p_value)

if p_value < alpha:
    print("Result: Reject the null hypothesis (H0)")
    print("Conclusion: There is a significant association between students' internet access at home and their desire to pursue higher education.")
else:
    print("Result: Fail to reject the null hypothesis (H0)")
    print("Conclusion: There is no significant association between the two variables.")


Chi-Square Statistic: 2.6941222240584533
P-Value: 0.10071894310832907
Result: Fail to reject the null hypothesis (H0)
Conclusion: There is no significant association between the two variables.


### Conclusions

#### How to interpret the Chi-Square Statistic:

Magnitude of the Chi-Square Statistic: The larger the Chi-Square Statistic, the greater the deviation between the observed data and what would be expected under the assumption of independence. A larger Chi-Square Statistic indicates stronger evidence against the null hypothesis.

The p-value associated with the chi-square test is 0.1007, which is greater than the chosen significance level (alpha = 0.05). When the p-value is greater than alpha, it indicates that we fail to reject the null hypothesis (H0).

# A summary of statistical tests and when to use each


| Statistical Test        | When to Use It               | Dataset Attributes                                            |
|-------------------------|------------------------------|--------------------------------------------------------------|
| t-Test (Independent Samples) | Compare means of two groups | - Continuous dependent variable<br> - Categorical independent variable (2 groups) |
| Paired t-Test           | Compare means of paired data | - Continuous dependent variable<br> - Paired or repeated measurements |
| Chi-Square Test of Independence | Assess association between categorical variables | - Categorical variables in a contingency table                |
| One-Way ANOVA           | Compare means of more than two groups | - Continuous dependent variable<br> - Categorical independent variable (with >2 levels) |
| Two-Way ANOVA           | Assess interaction effects between two categorical variables | - Continuous dependent variable<br> - Two categorical independent variables |
| Pearson's Correlation   | Examine linear relationship between two continuous variables | - Two continuous variables                                   |
| Spearman's Rank Correlation | Assess monotonic relationship between two variables | - Ordinal or continuous variables                            |
| Mann-Whitney U Test     | Compare distributions of two groups (non-parametric) | - Continuous dependent variable<br> - Categorical independent variable (2 groups) |
| Kruskal-Wallis Test     | Compare distributions of more than two groups (non-parametric) | - Continuous dependent variable<br> - Categorical independent variable (with >2 levels) |
| Linear Regression       | Predict a continuous outcome based on predictor variables | - Continuous dependent variable<br> - One or more predictor variables |
| Logistic Regression     | Predict a binary outcome based on predictor variables | - Binary dependent variable (0/1)<br> - One or more predictor variables |
